### PR TITLE
feat(logging): add configuration support for logger plugin

### PR DESCRIPTION
**Pull request by phantumcode**
## General Checklist

- [x] Added new tests to cover change, if needed
- [x] Build succeeds with all target using Swift Package Manager
- [x] All unit tests pass
- [x] All integration tests pass
- [x] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [x] Documentation update for the change if required
- [x] PR title conforms to conventional commit style
- [x] New or updated tests include `Given When Then` inline code documentation and are named accordingly `testThing_condition_expectation()`
- [x] If breaking change, documentation/changelog update with migration instructions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/Logging/Sources/AWSCloudWatchLoggingPlugin/Configuration/DefaultRemoteLoggingConstraintsProvider.swift [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+        return request
+    }
+    
+    func refresh() {
+        Task {
+            do {
+                _ = try await self.fetchLoggingConstraints()
+            } catch {
+                //TODO: log error
```
Comment (thisisabhash) [inline AmplifyPlugins/Logging/Sources/AWSCloudWatchLoggingPlugin/Configuration/DefaultRemoteLoggingConstraintsProvider.swift:121]: need to finish TODO before merging

File: AmplifyPlugins/Logging/Sources/AWSCloudWatchLoggingPlugin/Configuration/AWSCloudWatchLoggingPluginConfiguration.swift [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+   public let logGroupName: String
+   public let region: String
+   public let localStoreMaxSizeInMB: Int
+   public let flushIntervalInSeconds: Int
+   public let defaultRemoteConfiguration: DefaultRemoteConfiguration?
+   public let loggingConstraints: LoggingConstraints
+}
+
+import Amplify
+import AWSPluginsCore
```
Comment (thisisabhash) [inline AmplifyPlugins/Logging/Sources/AWSCloudWatchLoggingPlugin/Configuration/AWSCloudWatchLoggingPluginConfiguration.swift:40]: nit: import statements at top of the file

Comment (phantumcode) [reply AmplifyPlugins/Logging/Sources/AWSCloudWatchLoggingPlugin/Configuration/AWSCloudWatchLoggingPluginConfiguration.swift]: imports statements removed.

File: AmplifyPlugins/Logging/Sources/AWSCloudWatchLoggingPlugin/Configuration/DefaultRemoteLoggingConstraintsProvider.swift (hunk 1 above)
Comment (phantumcode) [reply AmplifyPlugins/Logging/Sources/AWSCloudWatchLoggingPlugin/Configuration/DefaultRemoteLoggingConstraintsProvider.swift]: adding logging for error.

**Review by thisisabhash — APPROVED** [review]
LGTM

**Pull request merged**